app/Services/Model/Imp/SummitTicketTypeService.php

```diff
@@ -350,6 +369,32 @@ public function getAllowedTicketTypes(Summit $summit, Member $member): array
                 )
             );
 
+            $promocode = null;
+
+            if (!is_null($filter)) {
+                $element = $filter->getUniqueFilter('promo_code');
+                $promocode_id = intval($element->getRawValue());
+                $promocode = $summit->getPromoCodeById($promocode_id);
+
+                if (is_null($promocode)) {
+                    throw new EntityNotFoundException(sprintf("Promo code id %s does not belongs to summit id %s.", $promocode_id, $summit->getId()));
+                }
+
+                if ($promocode->isPrepaid()) {
```

Comment:
this logic could be spilled  using a factory and 2 strategies 
one for pre paid
and one for regular

Reply:
done